lib/api.js

```diff
@@ -12,12 +12,18 @@ var
 
 module.exports = {
   request: function (https, api, req, data, responseObject, callback) {
-    request(
-      templates.apiSignature({
-        https: https,
-        api: api,
-        xml: templates[req](data)
-      }),
+    var xml = templates[req](data);
+    // Remove whitespace between XML elements
+    xml = xml.replace(/>\s*</mg, '><');
+
+    var protocol = https ? 'https://' : 'http://'; 
+    var url = protocol + 'secure.shippingapis.com/ShippingAPI.dll';
+    
+    request({
+        url: url,
+        method: 'GET',
+        qs: { API: api, XML: xml }
```

Comment:
Also wouldn't work without URI encoding all the stuff, so it seemed cleaner to use the builtin request functionality rather than a template.